src/code/z_msgevent.c

```diff
@@ -1,105 +1,1229 @@
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ * Command size: 1
+ */
+s32 MsgEvent_FocusToChild(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback, s32* endScript) {
+    if (actor->child != NULL) {
+        Actor_ChangeFocus(actor, play, actor->child);
+    }
+    return MSCRIPT_CONTINUE;
+}
+
+/**
+ * Sets player focus & talk actor to the actor executing the script
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ * Command size: 1
+ */
+s32 MsgEvent_FocusToSelf(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback, s32* endScript) {
+    if (actor->child != NULL) {
+        Actor_ChangeFocus(actor->child, play, actor);
+    }
+    return MSCRIPT_CONTINUE;
+}
+
+/**
+ * Unconditional branch by `skip`
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ *  1:(s16) skip
+ * Command size: 3
+ */
+s32 MsgEvent_Jump(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback, s32* endScript) {
+    u8* script = *scriptPtr;
+    s16 skip = MSCRIPT_GET_16(script, 1);
+
+    *scriptPtr += skip;
+    return MSCRIPT_CONTINUE;
+}
+
+/**
+ * Branch forwards if the specified quest item is obtained
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ *  1:(u16) questItem
+ *  3:(s16) skip
+ * Command size: 5
+ */
+s32 MsgEvent_BranchOnQuestItem(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback,
+                               s32* endScript) {
+    u8* script = *scriptPtr;
+    u16 questItem = MSCRIPT_GET_16(script, 1);
+    s16 skip = MSCRIPT_GET_16(script, 3);
+
+    if (CHECK_QUEST_ITEM(questItem)) {
+        *scriptPtr += skip;
+    }
+    return MSCRIPT_CONTINUE;
+}
+
+/**
+ * Branch forwards if event_inf flag is set
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ *  1:(u16) event_inf flag
+ *  3:(s16) skip
+ * Command size: 5
+ */
+s32 MsgEvent_BranchOnEventInf(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback,
+                              s32* endScript) {
+    u8* script = *scriptPtr;
+    s16 skip = MSCRIPT_GET_16(script, 3);
+
+    if (MSCRIPT_GET_8(script, 2) & gSaveContext.eventInf[MSCRIPT_GET_8(script, 1)]) {
+        *scriptPtr += skip;
+    }
+    return MSCRIPT_CONTINUE;
+}
+
+/**
+ * Sets event_inf flag
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ *  1:(u16) event_inf flag
+ * Command size: 3
+ */
+s32 MsgEvent_SetEventInf(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback, s32* endScript) {
+    u8* script = *scriptPtr;
+
+    gSaveContext.eventInf[MSCRIPT_GET_8(script, 1)] |= MSCRIPT_GET_8(script, 2);
+    return MSCRIPT_CONTINUE;
+}
+
+/**
+ * Unsets event_inf flag
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ *  1:(u16) event_inf flag
+ * Command size: 3
+ */
+s32 MsgEvent_UnsetEventInf(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback, s32* endScript) {
+    u8* script = *scriptPtr;
+
+    gSaveContext.eventInf[MSCRIPT_GET_8(script, 1)] &= MSCRIPT_GET_8(script, 2) ^ 0xFF;
```

Comment:
ohhh. That's a new way of unsetting eventInf.

Would it match like this? This is how the `CLEAR_EVENTINF` macro looks like
```suggestion
    gSaveContext.eventInf[MSCRIPT_GET_8(script, 1)] &= (u8)~MSCRIPT_GET_8(script, 2);
```

Reply:
Unfortunately it does not. Get a `nor` instead of an `xori`.